src/memory/AmdFlash.cc

```diff
@@ -43,14 +44,16 @@ AmdFlash::AmdFlash(const std::string& name, const ValidatedChip& validatedChip,
                    DeviceConfig& config, std::string_view id)
 	: motherBoard(config.getMotherBoard())
 	, chip(validatedChip.chip)
+	, syncOperation(motherBoard.getScheduler())
 {
 	cmd.reserve(5 + chip.program.bufferPageSize); // longest command is BufferProgram
 
 	assert(writeProtectSectors.size() <= chip.geometry.sectorCount);
 	sectors.reserve(chip.geometry.sectorCount);
 	for (size_t address = 0; const Region& region : chip.geometry.regions) {
-		for (size_t regionSector = 0; regionSector < region.count; ++regionSector, address += region.size) {
+		for (size_t regionSector = 0, index = 0; regionSector < region.count; ++regionSector, address += region.size, ++index) {
```

Comment:
Technically nothing wrong with this.
Personally I'm not a big fan of for loops that do multiple things (via the comma-operator) in the "next"-expression. Here you already do 3 different things. I don't think that's super readable.
But it's your code, pick the style that you prefer.


`regionSector` and `index` are always equal. (Almost certainly the compiler will figure this out and only use one variable). Do you think the intention of the code is more clearly expressed by using two variables? Or just one?

Reply:
Ai, I made a mistake here, index is supposed to be assigned to 0 in the top level. I want the total sector index.